### PR TITLE
[OF-1193] feat: Add XErrorCodeType

### PR DESCRIPTION
**Pull request by Mag-JB**
* Add XErrorCodeCatergory

connected-spaces-platform Pull Request

**For the reviewer**

* [ ] If required, are the changes covered by appropriate tests?
* [ ] Are any public-facing API changes well documented?
* [ ] Is the code easily readable and extensible and/or follow existing conventions?

### CODE REVIEW

File: Library/src/Systems/Quota/Quota.cpp [hunk 1]
```diff
@@ -367,4 +422,67 @@ const TierFeatures StringToTierFeatureEnum(const csp::common::String& Value)
 	return TierFeatures::SpaceOwner;
 }
 
+XErrorCodeTypes StringToXErrorCodeTypeEnum(const csp::common::String& Value)
+{
+	if (Value == "group_spaceownerquota")
```
Comment (MAG-AdamThorn) [inline Library/src/Systems/Quota/Quota.cpp:427]: The string case here does not match what is listed in the CHS TDD, is what you have here correct? Eg: **Group_SpaceOwnerQuota**

Comment (MAG-AdamThorn) [reply Library/src/Systems/Quota/Quota.cpp:427]: Same for the ones below

Comment (Mag-JB) [reply Library/src/Systems/Quota/Quota.cpp:427]: The TDD seems to be wrong

File: Library/src/Systems/Quota/Quota.cpp [hunk 2]
```diff
@@ -367,4 +422,67 @@ const TierFeatures StringToTierFeatureEnum(const csp::common::String& Value)
...
+		return XErrorCodeTypes::TicketedSpacesUserQuota;
+	}
+
+	CSP_LOG_ERROR_FORMAT("QuotaSystem XErrorCodeType not recognized: %s. Defaulting to Unknown", Value.c_str());
+	return XErrorCodeTypes::Unknown;
+}
+
+XErrorCategories StringToXErrorCatergoriesEnum(const csp::common::String& Value)
+{
+	if (Value == "quota_limitreached")
```
[the remark this answers is not in the document]
Comment (Mag-JB) [reply Library/src/Systems/Quota/Quota.cpp:478]: The TDD seems to be wrong

**Comment by Mag-JB** [comment]
Not auto merging until there is an answer about capitalisation

**Review by MAG-MichaelK — CHANGES_REQUESTED** [review]
We should be using the already available `FailureReason` value for exposing `X-ErrorCode` values, and completely ignoring the category as it doesn't matter to us. `ResultBase` already takes care of checking for the `X-ErrorCode` header for us. Please re-implement this by overriding `ParseErrorCode()` in the QuotaSystem result classes so that `ResultBase` can get an integer value representing the error from the string CHS gives us. For reference, please check https://github.com/magnopus-opensource/connected-spaces-platform/blob/develop/Library/src/Systems/Users/Authentication.cpp#L102

File: Library/src/Systems/Quota/Quota.cpp (hunk 2 above)
[the remark this answers is not in the document]
Comment (MAG-MichaelK) [reply Library/src/Systems/Quota/Quota.cpp:478]: We modify any incoming header value to be lowercase. CHS does indeed send it in the way Adam described, but that does not matter to us.

**Pull request closed without merging**